src/run.ts

```diff
@@ -95,11 +97,49 @@ function getDockerSecretArguments(secretName: string): string[] {
 
 function getGenericSecretArguments(secretName: string): string[] {
     const secretArguments = core.getInput('arguments');
+    const parsedArgument = fromLiteralsToFromFile(secretArguments);
     let args = ['create', 'secret', 'generic', secretName];
-    args.push(...argStringToArray(secretArguments));
+    args.push(...argStringToArray(parsedArgument));
     return args;
 }
 
+function fromLiteralsToFromFile(secretArguments: string): string {
+    const parsedArgument = secretArguments.split("--").reduce((argumentsBuilder, argument) => {
+        if (argument && !argument.startsWith("from-literal=")) argumentsBuilder += " --" + argument;
+        if (argument && argument.startsWith("from-literal=")) {
+            const command = argument.substring("from-literal=".length);
+            if (command.indexOf("=") == -1) throw new Error('Invalid from-literal input. It should contain a key and value');
+            const secretName = command.substring(0, command.indexOf("="));
+            const secretValue = command.substring(command.indexOf("=") + 1);
+            argumentsBuilder += " --from-file=" + createFile(path.join(process.env['RUNNER_TEMP'], secretName.trim()), secretValue);
+        }
+        return argumentsBuilder;
+    });
+    return parsedArgument;
+}
+
+function createFile(filePath: string, data: string, options?: any): string {
```

Comment:
If you're not supplying any options, there's no point to add it.

Reply:
Done